File: archive/src/job-boards/weworkremotely/board.js
import { BaseBoard } from '../base.js';

export class WeWorkRemotelyBoard extends BaseBoard {
  constructor(jobTitle) {
    super(jobTitle);
    this.searchUrl = `https://weworkremotely.com/remote-jobs/search?term=${encodeURIComponent(jobTitle)}`;
  }

  async extractJobListings(page) {
    // Wait for the job listings to load
    await page.waitForSelector('.jobs-container');

    // Extract job information
    return await page.evaluate(() => {
      const jobs = [];
      document.querySelectorAll('.job-listing').forEach(job => {
        const jobData = {
          position: job.querySelector('.title')?.textContent?.trim() || '',
          company: job.querySelector('.company')?.textContent?.trim() || '',
          location: job.querySelector('.location')?.textContent?.trim() || '',
          salary: job.querySelector('.salary')?.textContent?.trim() || '',
          tags: Array.from(job.querySelectorAll('.tags .tag')).map(tag => tag.textContent.trim()),
          posted: job.querySelector('.date')?.textContent?.trim() || '',
          description: job.querySelector('.description')?.textContent?.trim() || '',
          jobId: job.getAttribute('data-id') || ''
        };
        jobs.push(jobData);
      });
      return jobs;
    });
  }

  async processJob(job, browser) {
    const page = await browser.newPage();

    try {
      // Navigate to the job listing
      const jobUrl = `https://weworkremotely.com/remote-jobs/${job.jobId}`;
      await page.goto(jobUrl);

      // Wait for the content to load
      await page.waitForSelector('.job-description', { timeout: 10000 });

      // Get the full job description
      const fullDescription = await page.evaluate(() => {
        const desc = document.querySelector('.job-description');
        if (!desc) return '';
        // Remove any script tags
        Array.from(desc.querySelectorAll('script')).forEach(s => s.remove());
        return desc.textContent.trim();
      });

      // Extract additional job details
      const jobDetails = await page.evaluate(() => {
        const details = {};

        // Get posting time
        const timeElement = document.querySelector('.date');
        if (timeElement) {
          details.posted = timeElement.textContent.trim();
        }

        // Get company profile stats
        const companyProfile = document.querySelector('.company-profile');
        if (companyProfile) {
          const stats = companyProfile.querySelectorAll('.stat');
          stats.forEach(stat => {
            const label = stat.querySelector('.label')?.textContent?.trim().toLowerCase();
            const value = stat.querySelector('.value')?.textContent?.trim();
            if (label && value) {
              details[label] = value;
            }
          });
        }

        return details;
      });

      console.log(`\nJob Description Length: ${fullDescription.length} characters`);

      // Parse the description using Groq API
      const parsedDescription = await this.parseJobDescription(fullDescription);

      // Add the parsed description and additional details to the job data
      return {
        ...job,
        parsedDescription,
        jobUrl,
        posted: jobDetails.posted || '',
        views: jobDetails.views || '',
        applicants: jobDetails.applicants || '',
        applyPercentage: jobDetails.applyPercentage || ''
      };

    } catch (e) {
      console.error(`Error processing job ${job.position}: ${e.message}`);
      return null;
    } finally {
      await page.close();
    }
  }
} 

File: archive/python_version/job_boards/weworkremotely/board.py
from ..base import BaseJobBoard

class WeWorkRemotelyBoard(BaseJobBoard):
    def __init__(self, job_title):
        super().__init__(job_title)
        self.base_url = "https://weworkremotely.com"
        self.search_url = f"{self.base_url}/remote-jobs/search?term={job_title.replace(' ', '+')}"
    
    async def extract_job_listings(self, page):
        """Extract job listings from the search page."""
        return await page.evaluate('''() => {
            const jobs = [];
            document.querySelectorAll('.jobs article').forEach(job => {
                const title = job.querySelector('.title');
                const company = job.querySelector('.company');
                const location = job.querySelector('.location');
                const tags = Array.from(job.querySelectorAll('.tags .tag')).map(tag => tag.textContent.trim());
                
                if (title && company) {
                    jobs.push({
                        position: title.textContent.trim(),
                        company: company.textContent.trim(),
                        location: location ? location.textContent.trim() : 'Remote',
                        tags: tags,
                        job_id: job.getAttribute('data-id')
                    });
                }
            });
            return jobs;
        }''')
    
    async def process_job(self, job, playwright):
        """Process a single job listing."""
        browser = await playwright.chromium.launch(headless=False)
        page = await browser.new_page()
        
        try:
            # Navigate to the job listing
            job_url = f"{self.base_url}/remote-jobs/{job['job_id']}"
            await page.goto(job_url)
            
            # Wait for the content to load
            await page.wait_for_selector('.listing-container', timeout=10000)
            
            # Get the full job description
            full_description = await page.evaluate('''() => {
                const desc = document.querySelector('.listing-container');
                if (!desc) return '';
                return desc.textContent.trim();
            }''')
            
            # Extract additional job details
            job_details = await page.evaluate('''() => {
                const details = {};
                
                // Get posting date
                const dateElement = document.querySelector('.listing-header-container time');
                if (dateElement) {
                    details.posted = dateElement.getAttribute('datetime');
                }
                
                // Get company profile stats
                const companyProfile = document.querySelector('.company-profile');
                if (companyProfile) {
                    const stats = companyProfile.querySelectorAll('.stat');
                    stats.forEach(stat => {
                        const label = stat.querySelector('.label').textContent.trim();
                        const value = stat.querySelector('.value').textContent.trim();
                        details[label.toLowerCase()] = value;
                    });
                }
                
                return details;
            }''')
            
            print(f"\nJob Description Length: {len(full_description)} characters")
            
            # Save metadata
            metadata = {
                "job_id": job['job_id'],
                "position": job['position'],
                "company": job['company'],
                "location": job['location'],
                "tags": job['tags'],
                "job_url": job_url,
                "posted": job_details.get('posted', ''),
                "views": job_details.get('views', ''),
                "applicants": job_details.get('applicants', ''),
                "description_length": len(full_description)
            }
            self.save_metadata(job['job_id'], metadata)
            
            # Parse the description using Groq API
            parsed_description = await self.parse_job_description(full_description)
            
            # Save GROQ response
            self.save_groq_response(job['job_id'], parsed_description)
            
            # Add the parsed description and additional details to the job data
            job['parsed_description'] = parsed_description
            job['job_url'] = job_url
            job['posted'] = job_details.get('posted', '')
            job['views'] = job_details.get('views', '')
            job['applicants'] = job_details.get('applicants', '')
            
            # Save job listing
            self.save_job_listing(job['job_id'], job)
            
            return job
            
        except Exception as e:
            print(f"Error processing job {job['position']}: {str(e)}")
            return None
            
        finally:
            await browser.close() 